lib/cloud/clients.go

```diff
@@ -468,6 +495,21 @@ func (c *cloudClients) initAzureSubscriptionsClient() (*azure.SubscriptionClient
 	return client, nil
 }
 
+// initInstanceMetadata initializes the instance metadata client.
+func (c *cloudClients) initInstanceMetadata(ctx context.Context) (InstanceMetadata, error) {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+
```

Comment:
```suggestion
	if c.instanceMetadata != nil { // If some other thread already got here first.
		return c.instanceMetadata, nil
	}
```
need to check again since we unlocked the RLock before calling